camel/memories/blocks/graphdb_block.py

```diff
@@ -0,0 +1,224 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import logging
+from enum import Enum
+from typing import List, Optional
+
+from camel.embeddings import BaseEmbedding, OpenAIEmbedding
+from camel.memories.base import MemoryBlock
+from camel.memories.records import ContextRecord, MemoryRecord
+from camel.storages.graph_storages import BaseGraphStorage, Neo4jGraph
+from camel.types import OpenAIBackendRole, RoleType
+
+logger = logging.getLogger(__name__)
+
+
+class GraphDBBlock(MemoryBlock):
+    r"""An implementation of the MemoryBlock abstract base class for
+    maintaining and retrieving information using a graph database.
+
+    Args:
+        storage (Optional[BaseGraphStorage], optional): The storage mechanism
+            for the graph database. Defaults to a specific implementation
+            if not provided. (default: None)
+        embedding (Optional[BaseEmbedding], optional): Embedding mechanism
+            toconvert text into vector representations. Defaults to
+            OpenAIEmbedding if not provided. (default: None)
+    """
+
+    def __init__(
+        self,
+        storage: Optional[BaseGraphStorage] = None,
+        embedding: Optional[BaseEmbedding] = None,
+        url: Optional[str] = None,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        database: Optional[str] = "neo4j",
```

Comment:
these information should be passed to `BaseGraphStorage ` level rather than here